gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -119,12 +738,1491 @@ gum_stalker_add_call_probe (GumStalker * self,
+
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X16, ARM64_REG_X17);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X14, ARM64_REG_X15);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_STALKER_REG_CTX, ARM64_REG_X13);
+
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X4, ARM64_REG_X5);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X2, ARM64_REG_X3);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X0, ARM64_REG_X1);
+  }
+  else /* GUM_PROLOG_FULL */
+  {
+    gum_arm64_writer_put_pop_all_q_registers (cw);
+    gum_arm64_writer_put_pop_all_x_registers (cw);
+  }
+
+  /* restore the app_stack (with some tricks) */
+  gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X14, ARM64_REG_X15);
+  gum_arm64_writer_put_mov_reg_reg (cw, ARM64_REG_X14, ARM64_REG_SP);
+
+  gum_arm64_writer_put_ldr_reg_address (cw, ARM64_REG_X15,
+      GUM_ADDRESS (&ctx->app_stack));
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_X15, 0);
+  gum_arm64_writer_put_mov_reg_reg (cw, ARM64_REG_SP, ARM64_REG_X15);
+
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_X14, 8);
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X14, ARM64_REG_X14, 0);
+}
+
+static void
+gum_exec_ctx_write_push_branch_target_address (GumExecCtx * ctx,
+                                               const GumBranchTarget * target,
+                                               GumGeneratorContext * gc)
+{
+  GumArm64Writer * cw = gc->code_writer;
+
+#if DEBUG_LEVEL == 10
+  g_print ("gum_exec_ctx_write_push_branch_target_address - enter\n");
+  g_print ("\ttarget is %sdirect\n", (target->is_indirect) ? "in" : "");
+  g_print ("\ttarget->base is %svalid\n",
+      (target->base == X86_REG_INVALID) ? "in" : "");
+  g_print ("\ttarget->index  is %svalid\n",
+      (target->index == X86_REG_INVALID) ? "in" : "");
+#endif
+
+  if (!target->is_indirect)
+  {
+    if (target->base == ARM64_REG_INVALID)
+    {
+      gum_arm64_writer_put_ldr_reg_address (cw, ARM64_REG_X15,
+          GUM_ADDRESS (target->absolute_address));
+      gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X15, ARM64_REG_X15);
+    }
+    else
+    {
+      gum_exec_ctx_load_real_register_into (ctx, ARM64_REG_X15, target->base,
+          target->origin_ip, gc);
+      gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X15, ARM64_REG_X15);
+    }
+  }
+  else if (target->base == X86_REG_INVALID && target->index == X86_REG_INVALID)
+  {
+    g_assert ("not yet implemented"=="");
+  }
+  else
+  {
+    g_assert ("not yet implemented"=="");
+  }
+}
+
+static void
+gum_exec_ctx_load_real_register_into (GumExecCtx * ctx,
+                                      arm64_reg target_register,
+                                      arm64_reg source_register,
+                                      gpointer ip,
+                                      GumGeneratorContext * gc)
+{
+  if (gc->opened_prolog != GUM_PROLOG_MINIMAL) g_assert (""=="not implemented");
```

Comment:
`g_assert()` should go on the next line and just be indented by two spaces. Also, spaces missing around operator again.